capstone_project/src/test/java/com/google/sps/SchoolTest.java

```diff
@@ -58,10 +96,88 @@ public void getLatLongReturnsSchoolLatLong() {
   }
 
   @Test
-  public void getArticlesReturnsSchoolArticles() {
-    School uci = new School("UCI", uciLatitude, uciLongitude, uciNews);
-    ArrayList<String> actual = uci.getArticles();
-    ArrayList<String> expected = uciNews;
+  public void checkEqualsReturnsEqual() {
+    School uci1 = new School("UCI", uciLatitude, uciLongitude);
+    School uci2 = new School("UCI", uciLatitude, uciLongitude);
+    boolean actual = uci1.equals(uci2);
+    boolean expected = true;
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void checkEqualsReturnsNotEqual() {
+    School uci1 = new School("UCI", uciLatitude, uciLongitude);
+    School uci2 = new School("UCI", uciLatitude + 1, uciLongitude);
+    boolean actual = uci1.equals(uci2);
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void checkEqualsWorksForContains() {
+    School uci1 = new School("UCI", uciLatitude, uciLongitude);
+    School uci2 = new School("UCI", uciLatitude, uciLongitude);
+    ArrayList<School> schools = new ArrayList<School>();
+    schools.add(uci1);
+    schools.add(uci2);
+
+    boolean actual = schools.contains(uci1);
+    boolean expected = true;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void checkEqualsWorksForSet() {
+    School uci1 = new School("UCI", uciLatitude, uciLongitude);
+    School uci2 = new School("UCI", uciLatitude, uciLongitude);
+    School ucb = new School("UCB", 37.871942, -122.258476);
+    School ucla = new School("UCLA", 34.068965, -118.445245);
+    Set<School> schools = new HashSet<School>();
+    schools.add(uci1);
+    schools.add(uci2);
+    schools.add(ucb);
+    schools.add(ucla);
+    schools.add(ucla);
+
+    ArrayList<School> schoolsList = new ArrayList<School>(schools);
+    int actual = schoolsList.size();
+    int expected = 3;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testUploadingSchoolsWithNoDuplicatesInDatastore() {
+    when(request.getParameter("name-input")).thenReturn("UCB");
+    when(request.getParameter("latitude-input")).thenReturn("37.871942");
+    when(request.getParameter("longitude-input")).thenReturn("-122.258476");
+
+    try {
+      mapServlet.doPost(request, response);
+    } catch (IOException ex) {
+      System.out.println(ex.toString());
+    }
+
+    // Make sure that UCB was added, since it was not already in Datastore.
+    assertEquals(
+        2,
```

Comment:
nit: can you make variables for 2 to `expected` and datastore.prepare to `actual` same for below

Reply:
@Leoism these are all fixed now!